execution/executor/src/block_executor.rs

```diff
@@ -31,14 +31,14 @@ use aptos_vm::AptosVM;
 use fail::fail_point;
 use std::{marker::PhantomData, sync::Arc};
 
-pub trait TransactionBlockExecutor<T>: Send + Sync {
```

Comment:
If not for the different txn types, we don't need this trait from the first place, right? -- We just need to implement `VMExecutor` for the FakeExecutor.